src/main/java/duke2/Parser.java

```diff
@@ -0,0 +1,228 @@
+        return d1.format(DateTimeFormatter.ofPattern("MMM dd yyyy "));
+    }
+
+    /**
+     * Formats time to 12h
+     * 
+     * @param time to format
+     * @return formatted time in 12h
+     */
+    public static String timeFormat(String time) {
+        int i = Integer.parseInt(time);
+        if (i > 0000 && i < 1100) {
+            return i + "am";
+        } else if (i >= 1300 && i < 2359) {
+            char[] c = time.toCharArray();
+            String min = new StringBuilder().append(c[2]).append(c[3]).toString();
+            String hour = new StringBuilder().append(c[0]).append(c[1]).toString();
+
+            int h = Integer.parseInt(hour) - 12;
+            return String.valueOf(h) + min + "pm";
+        } else {
+            return i + "pm";
+        }
+
+    }
+
+    /**
+     * Parses string to create new task object
+     * 
+     * @param task to create
+     * @return new task from string
+     */
+    public static Task parseStringToTask(String task) {
+        String taskcopy = task;
+        String[] input = task.split(" ");
+        String[] inputcopy = taskcopy.split("\\[");
+
+        if (inputcopy[1].equals("E]")) {
+            String name = "";
+            String date = "";
+            for (int i = 1; i < input.length; i++) {
+                if (!input[i].equals("(at:")) {
+                    if (input[i + 1].equals("(at:")) {
+                        name += input[i];
+                        continue;
+                    }
+                    name += input[i] + " ";
+                } else {
+                    date += input[i + 1] + "-" + input[i + 2] + "-" + input[i + 3] + " " + input[i + 4];
+                    break;
+                }
+            }
+            return new Event(name, date);
+
+        } else if (inputcopy[1].equals("D]")) {
+            String name = "";
+            String date = "";
+            for (int i = 1; i < input.length; i++) {
+                if (!input[i].equals("(by:")) {
+                    if (input[i + 1].equals("(by:")) {
+                        name += input[i];
+                        continue;
+                    }
+                    name += input[i] + " ";
+                } else {
+                    date += input[i + 1] + "-" + input[i + 2] + "-" + input[i + 3] + " " + input[i + 4];
+                    break;
+                }
+            }
+            return new Deadline(name, date);
+
+        } else {
+            String name = "";
+            for (int i = 1; i < input.length; i++) {
+                name += input[i] + " ";
+            }
+            return new Todo(name);
+        }
+    }
+
+    /**
+     * Parses command inputted by user
+     * 
+     * @param command from user
+     * @param list    existing tasklist
+     * @param input   full input from user
+     * @param path    of file from storage
+     */
+    public void commandToTask(String command, ArrayList<Task> list, String[] input, String path) {
```

Comment:
Same goes for `commandToTask` method, it is really long and you should add some comments to explain the code, both for others and yourself. It will be a good practice to write comments to explain your code so that when next time you are revisiting this method, you can understand it easily.